packages/wallet/test/services/BalanceTracker.test.ts

```diff
@@ -47,4 +47,46 @@ describe('createBalanceTracker', () => {
       });
     });
   });
+
+  it('createBalanceByAddressTracker returns balance filtered by address', () => {
+    const address = utxo[0][1].address;
+    const utxoMultipleAddresses = utxo.slice(0, 3);
+
+    // Change the first utxo address so it's filtered out
+    utxoMultipleAddresses[0][1].address = Cardano.PaymentAddress('addr_test1qzs0umu0s2ammmpw0hea0w2crtcymdjvvlqngpgqy76gpfnuzcjqw982pcftgx53fu5527z2cj2tkx2h8ux2vxsg475qp3y3vz');
+    createTestScheduler().run(({ hot, expectObservable }) => {
+      const available$ = hot(       'a', { a: utxoMultipleAddresses }) as unknown as BehaviorObservable<Cardano.Utxo[]>;
+      const total$ = hot(           'a', { a: utxoMultipleAddresses }) as unknown as BehaviorObservable<Cardano.Utxo[]>;
+      const unspendable$ = hot(     'a', { a: utxoMultipleAddresses }) as unknown as BehaviorObservable<Cardano.Utxo[]>;
```

Comment:
A distinctUntilChanged is probably a good idea here, so that it doesn't emit if irrelevant utxo change
```suggestion
      const available$ = hot(       'aa', { a: utxoMultipleAddresses }) as unknown as BehaviorObservable<Cardano.Utxo[]>;
      const total$ = hot(           'aa', { a: utxoMultipleAddresses }) as unknown as BehaviorObservable<Cardano.Utxo[]>;
      const unspendable$ = hot(     'aa', { a: utxoMultipleAddresses }) as unknown as BehaviorObservable<Cardano.Utxo[]>;
```

Reply:
Fixed in [4bb58e4](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/4bb58e415d7e58c437939064515414896695ba6a)